xapian-letor/api/featurelist_internal.cc

```diff
@@ -0,0 +1,186 @@
+/** @file featurelist_internal.cc
+ * @brief Definition of Feature::Internal class.
+ */
+/* Copyright (C) 2012 Parth Gupta
+ * Copyright (C) 2016 Ayush Tomar
+ *
+ * This program is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 2 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program; if not, write to the Free Software
+ * Foundation, Inc., 51 Franklin St, Fifth Floor, Boston, MA  02110-1301
+ * USA
+ */
+
+#include <config.h>
+
+#include "xapian-letor/featurelist.h"
+#include "featurelist_internal.h"
+
+#include <cmath>
+#include <cstdio>
+#include <cstdlib>
+#include <cstring>
+#include "debuglog.h"
+
+using namespace std;
+using namespace Xapian;
+
+void
+FeatureList::Internal::set_database(const Xapian::Database & db)
+{
+    LOGCALL_VOID(API, "FeatureList::set_database", db);
```

Comment:
This is not `FeatureList::set_database`!

I'd just drop this ( - logging in both the API class method and the internal method it calls mostly just adds verbosity to the log, but if you are going to log here, use the correct method name and change the category to something appropriate (not `API`).

There are further instances of this below.